backend/utils/request_validation.py

```
@@ -0,0 +1,192 @@
+            if not request.is_json:
+                return jsonify({"error": "Content-Type must be application/json"}), 400
+
+            # Validate JSON structure
+            try:
+                data = request.get_json(force=True)
+                if not isinstance(data, dict):
+                    return (
+                        jsonify({"error": "Invalid JSON format. Expected object."}),
+                        400,
+                    )
+            except Exception as e:
```

Ruff (0.12.2)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)